client/txn.go

```diff
@@ -76,6 +76,7 @@ type Txn struct {
 	// is (still) unset.
 	haveTxnWrite bool
 	haveEndTxn   bool // True if there was an explicit EndTransaction
+	execing      bool // True if the txn is run in the exec() loop
```

Comment:
Yeah, that was my original attempt, but `kv/txn_correctness_text.go` uses `client.Txn.Commit` explicitly and requires different behavior (not aborting on commit failure) from transactions that are not part of the exec loop. I didn't dive deep enough into why. Want me to file an issue?